src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs

```diff
@@ -468,22 +469,23 @@ private void ProcessStaticResourceMarkupNode(object instance, XamlMemberDefiniti
 				string keyName = resourceNode.Members.FirstOrDefault()?.Value?.ToString();
 				var dependencyProperty = TypeResolver.FindDependencyProperty(member);
 
-				if (keyName != null && dependencyProperty != null)
+				if (keyName != null
+					&& dependencyProperty != null
+					&& instance is DependencyObject dependencyObject)
 				{
-					void ResolveResource()
-					{
-						object staticResource = ResolveStaticResource(instance, keyName);
+					ResourceResolver.ApplyResource(
+						dependencyObject,
+						dependencyProperty,
+						keyName,
+						isThemeResourceExtension: true);
```

Comment:
I'm a little bit confused.

The documentation of `isThemeResourceExtension` says `True for {ThemeResource Foo}, false for {StaticResource Foo}`

However, we're in `ProcessStaticResourceMarkupNode` method (which is actually called for both `ThemeResource` and `StaticResource`. Will that regress `StaticResource`?

Reply:
+1, this should check which it actually is.

Reply:
That's correct, I missed this!